modules/store/src/store.ts

```diff
@@ -175,20 +177,23 @@ export function select<
  * This overload is used to support spread operator with
  * fixed length tuples type in typescript 2.7
  */
-export function select<T, K = any>(
+export function select<State, Props = any, Result = any>(
+  propsOrPath: Props,
   ...paths: string[]
-): (source$: Observable<T>) => Observable<K>;
-export function select<T, K>(
-  pathOrMapFn: ((state: T) => any) | string,
+): (source$: Observable<State>) => Observable<Result>;
+export function select<State, Props, K>(
+  pathOrMapFn: (state: State, props?: Props) => any | string,
+  props: Props | string,
   ...paths: string[]
 ) {
-  return function selectOperator(source$: Observable<T>): Observable<K> {
+  return function selectOperator(source$: Observable<State>): Observable<K> {
     let mapped$: Observable<any>;
 
     if (typeof pathOrMapFn === 'string') {
-      mapped$ = source$.pipe(pluck(pathOrMapFn, ...paths));
+      const pathSlices = [<string>props, ...paths].filter(Boolean);
+      mapped$ = source$.pipe(pluck(pathOrMapFn, ...pathSlices));
     } else if (typeof pathOrMapFn === 'function') {
-      mapped$ = source$.pipe(map(pathOrMapFn));
+      mapped$ = source$.pipe(map(source => pathOrMapFn(source, <Props>props)));
```

Comment:
Why do `props` need to be typed here?

Reply:
Because the type was defined as `Props | string`. 
Therefor I had to cast it to `Props`.